lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3391,6 +3394,14 @@ namespace {
         func->setAccess(AccessLevel::Public);
       }
 
+      if (!isa<clang::CXXConstructorDecl>(decl) && !importedType) {
+        if (!Impl.importFunctionReturnType(decl, result->getDeclContext())) {
+          Impl.addImportDiagnostic(
+              decl, Diagnostic(diag::unsupported_return_type, decl),
+              decl->getSourceRange().getBegin());
+          return nullptr;
```

Comment:
Just for the sake of completeness, I'd like to mention that Objective-C and C will also hit this codepath, so there's a (hopefully very small?) chance of behavior change there.

Reply:
And here's an example:
```
struct alignas(128) OverAligned {
  char buff[128];
};

OverAligned test();
````